include/uapi/sound/sof/abi.h

```diff
@@ -31,6 +31,11 @@
 #define SOF_ABI_MINOR 23
 #define SOF_ABI_PATCH 0
 
+/* SOF ABI version major, minor and patch numbers for IPC4 */
+#define SOF_IPC4_ABI_MAJOR 4
+#define SOF_IPC4_ABI_MINOR 0
+#define SOF_IPC4_ABI_PATCH 0
```

Comment:
@libinyang I am not sure what/how we should interpret this. SOF SW has an incremental ABI with major 4 indicating an switch to use the new IPC4. Why would you introduce a new major/minor/patch for IPC4?

Reply:
> @libinyang I am not sure what/how we should interpret this. SOF FW has an incremental ABI with major 4 indicating a switch to use the new IPC4. Why would you introduce a new major/minor/patch for IPC4?

@ranj063 For IPC4, SOF driver can support both SOF FW and closed source FW. I'm a little confused that whether the ABI version in kernel means the SOF FW version in IPC4 mode? The kcontrol in the tplg can be used both for SOF FW and closed source FW. So how should we define the ABI version in kcontrol? Do you think we need to create a new ABI version dedicated for tplg kcontrol or reuse the SOF FW ABI?